gtest/src/mailbox.rs

```diff
@@ -357,4 +357,38 @@ mod tests {
 
         assert_eq!(system.balance_of(receiver_id), 1000);
     }
+
+    #[test]
+    fn delayed_message_works() {
+        // Arranging data for future messages
+        let system = System::new();
+        let message_id: MessageId = Default::default();
+        let source_user_id = ProgramIdWrapper::from(100).0;
+        let destination_user_id = ProgramIdWrapper::from(200).0;
+        let message_payload: Payload = vec![1, 2, 3].try_into().unwrap();
+        let log = Log::builder().payload(message_payload.clone());
+
+        // Building message based on arranged data
+        let message = Message::new(
+            message_id,
+            source_user_id,
+            destination_user_id,
+            message_payload.encode().try_into().unwrap(),
+            Default::default(),
+            0,
+            None,
+        );
+
+        let blocks = 10;
+        system.0.borrow_mut().send_delayed_dispatch(
+            blocks,
+            (message_id, Dispatch::new(DispatchKind::Handle, message)).into(),
+        );
+
+        let mailbox = system.get_mailbox(destination_user_id);
+        assert!(!mailbox.contains(&log));
+
+        system.spend_blocks(blocks);
```

Comment:
@LouiseMedova how do you see delayed processing while spending blocks? maybe it should return some info enum?

like

- spend_blocks called for 10 blocks
- in 3 blocks occurs delayed message, got processed, so spend block interrupts with returning RunResult + current bn maybe?

Reply:
I think it is not necessary to interrupt `spend_blocks`, and it's better to handle the `RunResult` when the  indicated number of blocks has passed